### ISSUE TITLE
Unable to run code generation

### ISSUE THREAD

**Issue by ilockett** [apollographql/apollo-ios#2543]
## Bug report

Installing Apollo 1.0.0 via Cocoapods results in code generation error.

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 1.0.0
- Xcode version: 14.0.1
- Swift version: 5.7
- Package manager: N/A
- Cocoapods version: 1.11.3

## Steps to reproduce

`pod init`

Add `pod 'Apollo'`

`pod install`

`./Pods/Apollo/apollo-ios-cli init --schema-name MySchema`
`./Pods/Apollo/apollo-ios-cli generate` 

```
ApolloCodegenLib/resource_bundle_accessor.swift:11: Fatal error: could not load resource bundle: from /Users/xxx/ProjectName/Pods/Apollo/Apollo_ApolloCodegenLib.bundle or /Users/xxx/Library/Caches/CocoaPods/Pods/Release/Apollo/1.0.0-95555/.build/arm64-apple-macosx/release/Apollo_ApolloCodegenLib.bundle
zsh: trace trap  ./Pods/Apollo/apollo-ios-cli generate
```

## Further details

Note that cocoapods is installed using Hombrew.

**Comment by arkulpaKonstantin** [comment]
Just ran into the same issue 😕

**Comment by calvincestari** [comment]
Thanks for the bug reports.

We made a fix for this type of error last week. There is also a test that's able to call the CLI and succeed. This is weird..

**Comment by calvincestari** [comment]
A quick thought - I wonder if it's related to not using the --path command, which means it's looking for the config file in the location of the CLI binary?

init didn't fail though so it seems it was able to write the config file to that location though.

Try this as a workaround - use an external path for both init and generate. See if that works because it then wouldn't be looking in the .Pods folder.

**Comment by ilockett** [comment]
> A quick thought - I wonder if it's related to not using the --path command, which means it's looking for the config file in the location of the CLI binary?
> 
> init didn't fail though so it seems it was able to write the config file to that location though.
> 
> Try this as a workaround - use an external path for both init and generate. See if that works because it then wouldn't be looking in the .Pods folder.

Thanks @calvincestari , unfortunately adding the `--path` to both `init` and `generate` resulted in the same error. I used a path deliberately not in the Pods folder.

**Comment by calvincestari** [comment]
Ok, I'll have to dig into it a bit later; we're all en route to Summit at the moment. Will try take a look later today. Apologies for the delay.

**Comment by indama** [comment]
Hi, The same issue here.

**Comment by ex4mple23** [comment]
Same problem. Can't sleep, the customer is rushing the work. I'm really looking forward to news about the fix.

**Comment by indama** [comment]
I tried installing via Package manager. When I add ApolloCodegenLib I have some errors. 
<img width="396" alt="Screenshot at Oct 04 16-32-27" src="https://user-images.githubusercontent.com/4445931/193809496-1dba87d2-20df-488a-8395-1257b49b314b.png">

**Comment by sammcode** [comment]
@calvincestari Hi there! Had a quick question regarding the errors @indama is hitting, as my team just hit the same problem. Is there a reason why ApolloCodeGenLib used to work for iOS? We set up an iOS project a few weeks ago and followed the documentation at the time that recommended using ApolloCodeGenLib over deprecated bash scripts for automatic code generation. I don't recall the docs mentioning macOS only anywhere, and it was working great for us on iOS until this week. We're trying to transition to the new Codegen CLI now.

**Comment by SilverTab** [comment]
I'll be honest; I find the instructions to get code generation to work in 1.0 to be very confusing. Looking at the available options listed in the documentation to install the codegen CLI: 

1. Swift PM: I thought this would work out for our project, but since we manage our Swift packages via Xcode, it doesn't seem like this solution applies? (If so, I haven't found out how).
2. Cocoapods: We don't use it so, not an option.
3. Manual install: Not a fan of the fact that the CLI will not be tied to the Apollo library version but figured I'd give it a try so that I can at least be unblocked. Followed the instructions and got this:

```
% git clone https://github.com/apollographql/apollo-ios.git
% cd apollo-ios
% make build-cli
swift build --product apollo-ios-cli -c release
error: 'apollo-ios': Invalid manifest
/Users/jeannicolas/Code/swift/apollo-ios/Package.swift:4:8: error: no such module 'PackageDescription'
import PackageDescription
       ^
make: *** [build-cli] Error 1```

Any help would be appreciated...

**Comment by sammcode** [comment]
We are also using Xcode to manage Swift packages, so we were also a bit confused on how to set that up. Specifically, it was unclear how we can edit the `apollo-codegen-configuration.json` file in Xcode. We made a little bit of progress with the SPM solution, but we are still running into issues (will go into detail below).

### Possible workaround for editing configuration file when using SPM
A workaround I tried with the SPM solution was opening the `apollo-codegen-configuration.json` file in a TextEditor rather than Xcode, which at least allowed us to edit the file.

<img src="https://user-images.githubusercontent.com/58298401/193900563-0f5e75db-85cd-45cf-9dae-690b02a02aea.png" width="600"> 

<img src="https://user-images.githubusercontent.com/58298401/193901015-a4c61c52-2996-437a-9b86-2507ffd742f5.png" width="600"> 

### Schema fetching errors
But when attempting to fetch the schema, we're still running into errors. 

<img src="https://user-images.githubusercontent.com/58298401/193901770-81dc6bb1-fc45-402e-93c3-5ba34a727309.png" width="600"> 

<img src="https://user-images.githubusercontent.com/58298401/193902606-4a473946-e061-426d-9f8e-4a5f3e04c2b8.png" width="900"> 

We've tried with two working graphql endpoints and have gotten the same error with both of them. We are unclear as to if there is a problem with our configuration, or if there is a bug of some sort here. 

The main error we keep getting is:
```
Error: A server with the specified hostname could not be found.
``` 
It also looks like its having an issue reading from a cache, as we get this error as well:
```
2022-10-04 14:37:29.452 apollo-ios-cli[31890:14972703] NetworkStorageDB:_openDBReadConnections: failed to open read connection to DB @ /Users/samuel.mcgarry/Library/Caches/apollo-ios-cli/Cache.db.  Error=14. Cause=unable to open database file
```

**Comment by m1entus** [comment]
I am getting similar issue when trying to run cli (get via Pods):
```
m1entus@MacBook-Pro geneva-ios % ./Pods/Apollo/apollo-ios-cli generate
ApolloCodegenLib/resource_bundle_accessor.swift:11: Fatal error: could not load resource bundle: from /Users/m1entus/ios/Pods/Apollo/Apollo_ApolloCodegenLib.bundle or /Users/m1entus/Library/Caches/CocoaPods/Pods/Release/Apollo/1.0.0-95555/.build/x86_64-apple-macosx/release/Apollo_ApolloCodegenLib.bundle
zsh: illegal hardware instruction  ./Pods/Apollo/apollo-ios-cli generate
```

**Comment by calvincestari** [comment]
> @calvincestari Hi there! Had a quick question regarding the errors @indama is hitting, as my team just hit the same problem. Is there a reason why ApolloCodeGenLib used to work for iOS?
> 
> We set up an iOS project a few weeks ago and followed the documentation at the time that recommended using ApolloCodeGenLib over deprecated bash scripts for automatic code generation. I don't recall the docs mentioning macOS only anywhere, and it was working great for us on iOS until this week. We're trying to transition to the new Codegen CLI now.

@sammcode I think there might be some confusion between ApolloCodegenLib and the new apollo-ios-cli. ApolloCodegenLib has always only been designed to run on macOS because code generation does not need to be executed on mobile devices. You can view the [ApolloCodegen class from `0.53.0`](https://github.com/apollographql/apollo-ios/blob/0.53.0/Sources/ApolloCodegenLib/ApolloCodegen.swift#L4) and it has the same macOS-only designation.

v1.0 moves away from recommending that code generation be executed with each build, which is what the bash scripting help before recommended. Note that you can still take that same approach but using the new apollo-ios-cli instead of the legacy Apollo tooling CLI; very similar names I'm sorry.

**Comment by calvincestari** [comment]
@SilverTab your make build error seems pretty odd; PackageDescription is a Swift Package Manager defined object so as long as the dependencies are fetched correctly it should all be there. Could you try resetting the package cache or dependencies and see if that helps resolve that error - `swift package reset`

**Comment by ibragimovscode1** [comment]
> > @calvincestari Hi there! Had a quick question regarding the errors @indama is hitting, as my team just hit the same problem. Is there a reason why ApolloCodeGenLib used to work for iOS?
> > We set up an iOS project a few weeks ago and followed the documentation at the time that recommended using ApolloCodeGenLib over deprecated bash scripts for automatic code generation. I don't recall the docs mentioning macOS only anywhere, and it was working great for us on iOS until this week. We're trying to transition to the new Codegen CLI now.
> 
Hi @calvincestari regarding to the issue we faced we going to downgrade to 0.53.0 version and continue working what you think ? Is it solid decision or Should we wait for a new Codegen CLI? 
Thanks.

**Comment by deltakroneker** [comment]
After upgrading to 1.0.0 i also encounter issue during the codegen phase:

`ApolloCodegenLib/resource_bundle_accessor.swift:11: Fatal error: could not load resource bundle: from /Users/XXXXXXXX/ios/Pods/Apollo/Apollo_ApolloCodegenLib.bundle or /Users/XXX/Library/Caches/CocoaPods/Pods/Release/Apollo/1.0.0-95555/.build/arm64-apple-macosx/release/Apollo_ApolloCodegenLib.bundle`

**Comment by AnthonyMDev** [comment]
Update: We know the root cause of the the issue with running Codegen when using CocoaPods. @calvincestari is working on the fix for that now, and we will have it released ASAP!

> I'll be honest; I find the instructions to get code generation to work in 1.0 to be very confusing. Looking at the available options listed in the documentation to install the codegen CLI:

@SilverTab Thank you for the feedback! We will work on trying to make the documentation clearer next week as well.

> Swift PM: I thought this would work out for our project, but since we manage our Swift packages via Xcode, it doesn't seem like this solution applies? (If so, I haven't found out how).

I'm looking into this now. We were under the impression that this would still work the same, but if we were wrong there, this is definitely an oversight on our side. We will have a resolution for you ASAP.

**Comment by m1entus** [comment]
@AnthonyMDev @calvincestari Thanks guys, can't wait for it! btw. If i am using multiple apollo schema endpoints, i am wondering if it possible to specify it in `apollo-codegen-configuration.json` and add `enum` namespace for each ?

**Comment by AnthonyMDev** [comment]
@SilverTab You can run the plugin commands using Xcode like this.
https://blog.eidinger.info/xcode-integration-of-swift-package-plugins-in-xcode-14

I'll update the docs to add this next week!

**Comment by AnthonyMDev** [comment]
@m1entus I'd love to talk more about this use case and understand what you want to achieve there.

**Comment by AnthonyMDev** [comment]
Looks like we need to explicitly support the use of the package plugin from within the Xcode project. This was an oversight on our side due to not knowing about this...

We'll get it working and update this thread as soon as we do!

**Comment by calvincestari** [comment]
Reopening this issue until we've fixed the Xcode project plugins workflow.

**Comment by calvincestari** [comment]
Anyone who's watching this issue for a fix to the original issue of `Fatal error: could not load resource bundle` when using CocoaPods; the fix (#2548) has been merged with so if you point your Pod to the `HEAD` of `main` it should work for you now.

We've got a couple other fixes we want to get into `main` before publishing a `1.0.1` release.

**Comment by indama** [comment]
> Reopening this issue until we've fixed the Xcode project plugins workflow.

I have one more question. I am working through documentation. After downloaded schema.json

I running this two command: 
./Pods/Apollo/apollo-ios-cli init --schema-name ${MySchemaName}
./Pods/Apollo/apollo-ios-cli generate

QUESTION: Is apallo-ios-cli automatically finds the schema.json file?


When I run generate function command I am getting error:

Error: Cannot query field "activate" on type "undefined"
.../Activate.graphql:2:5

Here is my .graphql file

mutation activate($phone: String!, $deviceId: String!, $deviceName: String!, $otp: String!, $trusted: Boolean!) {
    activate(phone: $phone, deviceId: $deviceId, deviceName: $deviceName, deviceType: 1, otp: $otp, trusted: $trusted) {
        errors {
            message
            code
            sec
        }
        message
        token
    }
}

**Comment by creednmd** [comment]
OK, got the schema downloading, but when I run the generate command I get another raft of errors:

`swift package --allow-writing-to-package-directory apollo-generate -v --path "Sources/GraphQLClient/apollo-codegen-config.json"`

```
Building for debugging...
[196/196] Linking apollo-ios-cli
Build complete! (2.92s)
[DEBUG - ApolloCodegenLib:Glob.swift:116] - Expanding globstar **/*.graphqls
[DEBUG - ApolloCodegenLib:Glob.swift:180] - Expanded to /Users/myname/dev/pixplay/Sources/GraphQLClient/*.graphqls
[DEBUG - ApolloCodegenLib:Glob.swift:194] - Matching /Users/myname/dev/pixplay/Sources/GraphQLClient/*.graphqls
[DEBUG - ApolloCodegenLib:Glob.swift:116] - Expanding globstar **/*.graphql
[DEBUG - ApolloCodegenLib:Glob.swift:180] - Expanded to /Users/myname/dev/pixplay/Sources/GraphQLClient/*.graphql
[DEBUG - ApolloCodegenLib:Glob.swift:194] - Matching /Users/myname/dev/pixplay/Sources/GraphQLClient/*.graphql
[DEBUG - ApolloCodegenLib:Glob.swift:208] - Matched /Users/myname/dev/pixplay/Sources/GraphQLClient/schema.graphql
ApolloCodegenLib/JavaScriptBridge.swift:343: Precondition failed: Expected JavaScript array but found: undefined
```

**Comment by creednmd** [comment]
```
{
  "schemaName" : "myScheme",
  "input" : {
    "operationSearchPaths" : [
      "**/*.graphql"
    ],
    "schemaSearchPaths" : [
      "**/*.graphqls"
    ]
  },
  "output" : {
    "testMocks" : {
      "none" : {
      }
    },
    "schemaTypes" : {
      "path" : "./Sources/GraphQLClient",
      "moduleType" : {
        "swiftPackageManager" : {
        }
      }
    },
    "operations" : {
      "relative" : {
      }
    }
  },
  "schemaDownloadConfiguration": {
      "downloadMethod": {
          "introspection": {
              "endpointURL": "https://example.com/graphql",
              "httpMethod": {
                  "POST": {}
              },
              "includeDeprecatedInputValues": false,
              "outputFormat": "SDL"
          }
      },
      "downloadTimeout": 60,
      "headers": [
          { "key": "x-api-key", "value": "DEADBEEF" },
          { "key": "Authorization", "value": "Bearer DEADBEEF" }
      ],
      "outputPath": "./Sources/GraphQLClient/schema.graphql"
  }
}

```

**Comment by elviva404** [comment]
@creednmd Thank you so much, fetching the schema with a disabled sandbox also resolved my issue.

**Comment by AnthonyMDev** [comment]
It looks like in order to make network requests, the CLI may need to be run with the sandbox disabled. This was not clear to us from Apple’s (very limited) documentation on the switch package plug-in API. We’re going to do some more digging and update the documentation. Thanks do much to everyone who has helped dig into this issue so far!

I’m going to look into if there is a better/easier way for use to deploy the CLI. We felt that a swift package plug-in was the most user friendly way to do this, but it’s looking like the SPM plug-in system is still not mature enough to provide an experience here that is up to our standards.

I genuinely apologize to all of you who have started to attempt the migration to what we felt was a stable 1.0 version and have hit these issues. We’re going to spend time this coming week figuring out what the right option is for making code generation as seamless as possible for all of you.

**Comment by creednmd** [comment]
Looks like my generate errors where self-inflicted, and due to one or two paths being wrong, such that it was not finding any graphQL files, and passing an empty array; maybe a more accurate error than 'Precondition failed: Expected JavaScript array but found: undefined' would be in order. But despite all the trouble, have got it all migrated and working again.

**Comment by calvincestari** [comment]
That's great to read @creednmd, glad it's now working for you again. The CLI `init` command creates the config file with defaults that are very basic. So depending on your specific project setup you may need to adjust those values. Maybe that's a message worth adding to the end of the `init` command output.

**Comment by calvincestari** [comment]
Once #2554 is merged I believe we've fixed all the root causes in this issue, the others seem to be user configuration related. Maybe we can get in another one or two other issues before publishing `1.0.1`.

**Comment by elviva404** [comment]
> > Reopening this issue until we've fixed the Xcode project plugins workflow.
> 
> I have one more question. I am working through documentation. After downloaded schema.json
> 
> I am running this two command: ./Pods/Apollo/apollo-ios-cli init --schema-name ${MySchemaName} ./Pods/Apollo/apollo-ios-cli generate
> 
> QUESTION: Is apallo-ios-cli automatically finds the schema.json file?
> 
> When I run generate function command I am getting error:
> 
> Error: Cannot query field "activate" on type "undefined" .../Activate.graphql:2:5
> 
> Here is my .graphql file
> 
> > mutation activate($phone: String!, $deviceId: String!, $deviceName: String!, $otp: String!, $trusted: Boolean!) {
> > activate(phone: $phone, deviceId: $deviceId, deviceName: $deviceName, deviceType: 1, otp: $otp, trusted: $trusted) {
> > errors {
> > message
> > code
> > sec
> > }
> > message
> > token
> > }
> > }

Hi, I also have an issue similar to this please. 

```
Error: Cannot query field "company" on type "undefined"

/{PATH}/ExampleQuery.graphql:2:3
1 | query Query {
2 |   company {
  |   ^
3 |     ceo
```

My Query file `ExampleQuery.graphql`

```
query Query {
  company {
    ceo
  }
  roadster {
    apoapsis_au
  }
}
```

**Comment by elviva404** [comment]
@AnthonyMDev can you kindly assist me with the issue above.

**Comment by AnthonyMDev** [comment]
@elviva404 It looks like something is wrong with your schema here. It might be that you have named the query `Query`, but I don't think that should be an issue. Try changing the name to `MyQuery` or something, just to see if that's the issue.

In your `apollo-codegen-config.json` file, make sure that you are pointing to the path of your schema correctly. Let me know if that's your problem, because we should be emitting a better error there if so.

**Comment by elviva404** [comment]
Thanks

**Comment by AnthonyMDev** [comment]
@elviva404 Please keep me updated. Want to make sure you're able to get this working!

**Comment by ilockett** [comment]
> **This issue has been closed as the specific cause of it has been corrected. The fix for this will be released in 1.0.1 very soon and is available right now on the `main` branch**.
> 
> If you are having other problems with codegen, please submit a new issue!
> 
> Thanks to everyone for all the info as we work this out. We unfortunately did not get enough users during the Beta period to find these edge cases. So we are working quickly to resolve all of them now and will release a patch version ASAP!

Thanks for the update @AnthonyMDev - appreciate you and @calvincestari replying to us all and sorting these issues so quickly.

**Comment by DavidFerreiraYEO** [comment]
Hi, @AnthonyMDev  I'm getting a similar error as @elviva404:

I'm trying to generate code following this [steps][1] using cocoa pods.
On step: 5. Setup and run code generation using `appolo-ios-cli generate` command I'm getting this error:

```
 Error: Cannot query field "getAuthServiceHealth" on type "undefined"
 ./../NetworkInterface/GraphQL/QueriesList.graphql:2:2
1 | query Health {
2 | 	getAuthServiceHealth{
  |  ^
3 | 		status
```
Queries.graphql:

```
query Health {
	getAuthServiceHealth{
		status
		service
	}
}
```
schema:
```
{
  "__schema": {
    "queryType": {
      "name": "Query"
    },
    "mutationType": null,
    "subscriptionType": null,
    "types": [
      {
        "kind": "OBJECT",
        "name": "Query",
        "description": null,
        "fields": [
          {
            "name": "getAuthServiceHealth",
            "description": null,
            "args": [],
            "type": {
              "kind": "NON_NULL",
              "name": null,
              "ofType": {
                "kind": "OBJECT",
                "name": "HealthResponse",
                "ofType": null
              }
            },
            "isDeprecated": false,
            "deprecationReason": null
          },
```

Why I'm getting this error?

Thanks


  [1]: https://www.apollographql.com/docs/ios/get-started

**Comment by calvincestari** [comment]
@DavidFerreiraYEO, that looks like a GraphQL syntax error. Have you confirmed that your query works in Apollo Studio or GraphiQL?

If it works there it should work for code generation. If not, get it working there first.

**Comment by DavidFerreiraYEO** [comment]
![Screenshot 2022-10-14 at 15 43 21](https://user-images.githubusercontent.com/115166927/195874736-6922b39b-c55f-4cf2-b907-c786b98618bf.png)

It works. :/

**Comment by DavidFerreiraYEO** [comment]
@calvincestari @AnthonyMDev I just re-do everything in a clean project:

steps I done: 
- pod install (last version 1.0.1)
- apollo schema:download --endpoint=https://xxxxxx/graphql myUniqueSchema.json
- ./Pods/Apollo/apollo-ios-cli init --schema-name myUniqueSchema.json
- ./Pods/Apollo/apollo-ios-cli generate

now I'm getting different output: 
```
Error: GraphQLSchemaValidationError: Must provide only one schema definition.

There can be only one query type in schema.

There can be only one mutation type in schema.

There can be only one subscription type in schema.

There can be only one type named "Query".

Field "Query.hero" can only be defined once.

Field "Query.reviews" can only be defined once.

Field "Query.search" can only be defined once.

Field "Query.character" can only be defined once.

Field "Query.droid" can only be defined once.

Field "Query.human" can only be defined once.

Field "Query.starship" can only be defined once.

There can be only one type named "Mutation".

Field "Mutation.createReview" can only be defined once.

There can be only one type named "Subscription".

Field "Subscription.reviewAdded" can only be defined once.

There can be only one type named "Episode".

Enum value "Episode.NEWHOPE" can only be defined once.

Enum value "Episode.EMPIRE" can only be defined once.

Enum value "Episode.JEDI" can only be defined once.

There can be only one type named "Character".

Field "Character.id" can only be defined once.

Field "Character.name" can only be defined once.

Field "Character.friends" can only be defined once.

Field "Character.friendsConnection" can only be defined once.

Field "Character.appearsIn" can only be defined once.

There can be only one type named "LengthUnit".

Enum value "LengthUnit.METER" can only be defined once.

Enum value "LengthUnit.FOOT" can only be defined once.

There can be only one type named "Human".

Field "Human.id" can only be defined once.

Field "Human.name" can only be defined once.

Field "Human.homePlanet" can only be defined once.

Field "Human.height" can only be defined once.

Field "Human.mass" can only be defined once.

Field "Human.friends" can only be defined once.

Field "Human.friendsConnection" can only be defined once.

Field "Human.appearsIn" can only be defined once.

Field "Human.starships" can only be defined once.

There can be only one type named "Droid".

Field "Droid.id" can only be defined once.

Field "Droid.name" can only be defined once.

Field "Droid.friends" can only be defined once.

Field "Droid.friendsConnection" can only be defined once.

Field "Droid.appearsIn" can only be defined once.

Field "Droid.primaryFunction" can only be defined once.

There can be only one type named "FriendsConnection".

Field "FriendsConnection.totalCount" can only be defined once.

Field "FriendsConnection.edges" can only be defined once.

Field "FriendsConnection.friends" can only be defined once.

Field "FriendsConnection.pageInfo" can only be defined once.

There can be only one type named "FriendsEdge".

Field "FriendsEdge.cursor" can only be defined once.

Field "FriendsEdge.node" can only be defined once.

There can be only one type named "PageInfo".

Field "PageInfo.startCursor" can only be defined once.

Field "PageInfo.endCursor" can only be defined once.

Field "PageInfo.hasNextPage" can only be defined once.

There can be only one type named "Review".

Field "Review.episode" can only be defined once.

Field "Review.stars" can only be defined once.

Field "Review.commentary" can only be defined once.

There can be only one type named "ReviewInput".

Field "ReviewInput.stars" can only be defined once.

Field "ReviewInput.commentary" can only be defined once.

Field "ReviewInput.favorite_color" can only be defined once.

There can be only one type named "ColorInput".

Field "ColorInput.red" can only be defined once.

Field "ColorInput.green" can only be defined once.

Field "ColorInput.blue" can only be defined once.

There can be only one type named "Starship".

Field "Starship.id" can only be defined once.

Field "Starship.name" can only be defined once.

Field "Starship.length" can only be defined once.

Field "Starship.coordinates" can only be defined once.

There can be only one type named "SearchResult".

Must provide only one schema definition.

There can be only one query type in schema.

There can be only one mutation type in schema.

There can be only one subscription type in schema.

There can be only one type named "Query".

Field "Query.hero" can only be defined once.

Field "Query.reviews" can only be defined once.

Field "Query.search" can only be defined once.

Field "Query.character" can only be defined once.

Field "Query.droid" can only be defined once.

Field "Query.human" can only be defined once.

Field "Query.starship" can only be defined once.

There can be only one type named "Mutation".

Field "Mutation.createReview" can only be defined once.

There can be only one type named "Subscription".

Field "Subscription.reviewAdded" can only be defined once.

There can be only one type named "Episode".

Enum value "Episode.NEWHOPE" can only be defined once.

Enum value "Episode.EMPIRE" can only be defined once.

Enum value "Episode.JEDI" can only be defined once.

There can be only one type named "Character".

Field "Character.id" can only be defined once.

Field "Character.name" can only be defined once.

Field "Character.friends" can only be defined once.

Field "Character.friendsConnection" can only be defined once.

Field "Character.appearsIn" can only be defined once.

There can be only one type named "LengthUnit".

Enum value "LengthUnit.METER" can only be defined once.

Enum value "LengthUnit.FOOT" can only be defined once.

There can be only one type named "Human".

Field "Human.id" can only be defined once.

Field "Human.name" can only be defined once.

Field "Human.homePlanet" can only be defined once.

Field "Human.height" can only be defined once.

Field "Human.mass" can only be defined once.

Field "Human.friends" can only be defined once.

Field "Human.friendsConnection" can only be defined once.

Field "Human.appearsIn" can only be defined once.

Field "Human.starships" can only be defined once.

There can be only one type named "Droid".

Field "Droid.id" can only be defined once.

Field "Droid.name" can only be defined once.

Field "Droid.friends" can only be defined once.

Field "Droid.friendsConnection" can only be defined once.

Field "Droid.appearsIn" can only be defined once.

Field "Droid.primaryFunction" can only be defined once.

There can be only one type named "FriendsConnection".

Field "FriendsConnection.totalCount" can only be defined once.

Field "FriendsConnection.edges" can only be defined once.

Field "FriendsConnection.friends" can only be defined once.

Field "FriendsConnection.pageInfo" can only be defined once.

There can be only one type named "FriendsEdge".

Field "FriendsEdge.cursor" can only be defined once.

Field "FriendsEdge.node" can only be defined once.

There can be only one type named "PageInfo".

Field "PageInfo.startCursor" can only be defined once.

Field "PageInfo.endCursor" can only be defined once.

Field "PageInfo.hasNextPage" can only be defined once.

There can be only one type named "Review".

Field "Review.episode" can only be defined once.

Field "Review.stars" can only be defined once.

Field "Review.commentary" can only be defined once.

There can be only one type named "ReviewInput".

Field "ReviewInput.stars" can only be defined once.

Field "ReviewInput.commentary" can only be defined once.

Field "ReviewInput.favorite_color" can only be defined once.

There can be only one type named "ColorInput".

Field "ColorInput.red" can only be defined once.

Field "ColorInput.green" can only be defined once.

Field "ColorInput.blue" can only be defined once.

There can be only one type named "Starship".

Field "Starship.id" can only be defined once.

Field "Starship.name" can only be defined once.

Field "Starship.length" can only be defined once.

Field "Starship.coordinates" can only be defined once.

There can be only one type named "SearchResult". 
```

This fields are nothing related to my schema. 
I'm sure I have only one schema.

**Comment by calvincestari** [comment]
@DavidFerreiraYEO I've seen that error when codegen ends up finding all the test schemas we use in the repo. We have https://github.com/apollographql/apollo-ios/issues/2552 to fix that behaviour. The workaround for now is to put your schema and operation definition files in a subfolder and not the root of your project.

An example is putting them in a subfolder off the root named `graphql`, and then the `schemaSearchPaths` and `operationSearchPaths` could be `./graphql/**/*.graphqls` and `./graphql/**/*.graphql` respectively.

**Comment by DavidFerreiraYEO** [comment]
@calvincestari I create the folder graphql.
I put inside the schema and operations inside:
And I got again same query error:
```
Error: Cannot query field "getAuthServiceHealth" on type "undefined"
./graphql/Queries.graphql:2:2
1 | query Health {
2 | 	getAuthServiceHealth{
  |  ^
3 | 		status```

**Comment by AnthonyMDev** [comment]
You have a schema json file so your schema search path should be `./graphql/**/*.json`.

**Comment by DavidFerreiraYEO** [comment]
Hi @AnthonyMDev, 

that's the reason. Thanks for your help.

**Comment by elviva404** [comment]
> @elviva404 Please keep me updated. Want to make sure you're able to get this working!

Hi @AnthonyMDev it works now 💃🏾.

**Comment by sam961** [comment]
I followed the same steps ..
All are good not but the generated output doesn't include 
the swift classes
Attached are screenshots
![Screen Shot 2022-10-24 at 11 03 38 AM](https://user-images.githubusercontent.com/4921874/197477535-0aed3557-0c4a-4083-97fc-5561c9bfa4ea.png)

![Screen Shot 2022-10-24 at 11 03 19 AM](https://user-images.githubusercontent.com/4921874/197477608-78262f18-f85a-4daa-b128-75853068a3dd.png)
![Screen Shot 2022-10-24 at 11 03 49 AM](https://user-images.githubusercontent.com/4921874/197477616-3100793a-14a3-4fb0-9999-600b75b45e53.png)
![Screen Shot 2022-10-24 at 11 03 29 AM](https://user-images.githubusercontent.com/4921874/197477672-badbb6b1-f965-421b-a41c-40f97e4082ad.png)


What could be the issue ?

**Comment by tlester** [comment]
@AnthonyMDev @calvincestari Is there anything we can do to get v1.x working?  Or is it time to punt and go back to the .59 release?   I need to move on, but can't really be beta testing this right now.   

I *think* I got the init and fetch phases done via the cli.   But, the generate, is crashing hard (`[1]    85763 trace trap  apollo-ios-cli generate -v`).  

I just need good avice here.  If there is a work around, I'm all for it... if not, I need to know so I can downgrade and move on.  Is there any downside to the downgrade?  Is it just the codegen that's different?  Or is the 1.0 client better (more peformant, more stable, etc)?

**Comment by calvincestari** [comment]
@sam961 I think your issues might be caused by the `schemaName` you're using in the config file. That name is used for namespacing and import statements; `test.json` seems like the filename of your actual schema file and not what you want your code generated schema to be named. Use another name and try again.

**Comment by calvincestari** [comment]
@tlester, answers below

> Is there anything we can do to get v1.x working? Or is it time to punt and go back to the .59 release?

I'd need to know more about what your specific issues are. I see that you reacted to [this comment](https://github.com/apollographql/apollo-ios/issues/2543#issuecomment-1267247793) so if your error is the same as what SilverTab was experiencing then I believe it was [resolved for him by resetting the package cache](https://github.com/apollographql/apollo-ios/issues/2543#issuecomment-1268446856).

> I think I got the init and fetch phases done via the cli. But, the generate, is crashing hard ([1] 85763 trace trap apollo-ios-cli generate -v).

Is there any other error message?

> I just need good avice here. If there is a work around, I'm all for it... if not, I need to know so I can downgrade and move on. Is there any downside to the downgrade? Is it just the codegen that's different? Or is the 1.0 client better (more peformant, more stable, etc)?

Once we understand more about your specific problems we can help. This particular GitHub issue has become clouded with too many individual threads. I believe we've addressed the bugs related to this original issue in `1.0.1` and `1.0.2`; the rest appear to be configuration related.

The generated code in 1.x has many improvements. The 1.x documentation can certainly be improved to offer better guidance and self-serve many of the problems reported. `0.53.0` also has limited support at the moment and an end-of-life date will be announced soon.

**Comment by calvincestari** [comment]
For anyone watching this issue and dealing with error responses that look like
> Error: Cannot query field "company" on type "undefined"

https://github.com/apollographql/apollo-ios/issues/2567 has been merged and has better error responses when the schema or operation files cannot be found in the search paths. That was the root of that issue.

**Comment by jhiat00** [comment]
> We are also using Xcode to manage Swift packages, so we were also a bit confused on how to set that up. Specifically, it was unclear how we can edit the `apollo-codegen-configuration.json` file in Xcode. We made a little bit of progress with the SPM solution, but we are still running into issues (will go into detail below).
> 
> ### Possible workaround for editing configuration file when using SPM
> A workaround I tried with the SPM solution was opening the `apollo-codegen-configuration.json` file in a TextEditor rather than Xcode, which at least allowed us to edit the file.
> 
> <img alt="" width="600" src="https://user-images.githubusercontent.com/58298401/193900563-0f5e75db-85cd-45cf-9dae-690b02a02aea.png">
> 
> <img alt="" width="600" src="https://user-images.githubusercontent.com/58298401/193901015-a4c61c52-2996-437a-9b86-2507ffd742f5.png">
> 
> ### Schema fetching errors
> But when attempting to fetch the schema, we're still running into errors.
> 
> <img alt="" width="600" src="https://user-images.githubusercontent.com/58298401/193901770-81dc6bb1-fc45-402e-93c3-5ba34a727309.png">
> 
> <img alt="" width="900" src="https://user-images.githubusercontent.com/58298401/193902606-4a473946-e061-426d-9f8e-4a5f3e04c2b8.png">
> 
> We've tried with two working graphql endpoints and have gotten the same error with both of them. We are unclear as to if there is a problem with our configuration, or if there is a bug of some sort here.
> 
> The main error we keep getting is:
> 
> ```
> Error: A server with the specified hostname could not be found.
> ```
> 
> It also looks like its having an issue reading from a cache, as we get this error as well:
> 
> ```
> 2022-10-04 14:37:29.452 apollo-ios-cli[31890:14972703] NetworkStorageDB:_openDBReadConnections: failed to open read connection to DB @ /Users/samuel.mcgarry/Library/Caches/apollo-ios-cli/Cache.db.  Error=14. Cause=unable to open database file
> ```

@sammcode  were you able to get past this issue? I am currently having the same problem.

**Comment by AnthonyMDev** [comment]
The schema fetching errors are because the SPM plugin is in a sandbox and won't allow the disk read/write and networking mechanisms of the fetch schema command to work properly. You can try passing the `--disable-sandbox` flag, but if that doesn't work, we recommend installing the CLI manually.

Version 1.0.4 which is coming soon removes the existing SPM plugins and instead has a single plugin command that installs the CLI, you can then use the CLI from the command line and all these SPM plugin permissions issues go away!

**Comment by phea3** [comment]
it said : Error: A GraphQL schema could not be found. Please verify the schema search paths.
<img width="1080" alt="Screenshot 2022-11-17 at 11 32 39 AM" src="https://user-images.githubusercontent.com/98304098/202356114-98248fa2-be95-40ff-9856-81f6a8e010f5.png">

**Comment by elviva404** [comment]
Hi @AnthonyMDev I replaced `moduleType` from `packageManager` to `embeddedInTarget` and when I later ran code generation, the generated files were empty.  However when I switch back to `packageManager`, the schema package is populated with files/code accordingly.

 is there something I have done wrong in my setup? a prompt response would be appreciated please.
 
<img width="1311" alt="Screenshot 2022-11-29 at 7 32 30 PM" src="https://user-images.githubusercontent.com/33577593/204630823-a38f43c7-48b6-45f4-9958-90298282156e.png">

<img width="846" alt="Screenshot 2022-11-29 at 7 30 09 PM" src="https://user-images.githubusercontent.com/33577593/204630810-1b6fa7d4-2502-4636-8f6a-c33db798c272.png">

**Comment by AnthonyMDev** [comment]
@elviva404 I've never heard reports of us generating empty files before. Try just deleting the `GenSchema` folder completely and then re-running code generation from a clean state with `embeddedInTarget`. With the config you provided, all your models should go into that folder.

**Comment by AnthonyMDev** [comment]
I am locking this issue now. The original problem has been resolved, and people are using this closed issue as a catch-all for debugging multiple different configuration errors with code gen. This means everyone who has ever commented here (and has had their issue resolved) continues to get notifications.

**If you have an problem with getting code generation working, please open a new issue.**

Thank you.
